Could not find a declaration file for module 'http-cache-semantics'

**Describe the bug**
When using `cacheable-request@10.2.0` in my TypeScript project, I get the following error when trying to compile it:
```
maxim@ubuntu:~/google-api-typings-generator$ tsc
node_modules/cacheable-request/dist/types.d.ts:7:35 - error TS7016: Could not find a declaration file for module 'http-cache-semantics'. '/home/maxim/google-api-typings-generator/node_modules/http-cache-semantics/index.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/http-cache-semantics` if it exists or add a new declaration (.d.ts) file containing `declare module 'http-cache-semantics';`

7 import { CachePolicyObject } from 'http-cache-semantics';
                                    ~~~~~~~~~~~~~~~~~~~~~~


Found 1 error in node_modules/cacheable-request/dist/types.d.ts:7
```

**How To Reproduce (best to provide workable code or tests!)**
The problem is currently present in this PR: https://github.com/Maxim-Mazurok/google-api-typings-generator/pull/709

1. `git clone -b create-pr-action/chore-npm-dedupe-0 https://github.com/Maxim-Mazurok/google-api-typings-generator`
2. `cd google-api-typings-generator`
3. optional `nvm i`
4. `npm ci`
5. Observe the error

--- a/package.json
+++ b/package.json
@@ -33,6 +33,7 @@
 		"compliant"
 	],
 	"dependencies": {
+		"@types/http-cache-semantics": "^4.0.1",
 		"get-stream": "^6.0.1",
 		"http-cache-semantics": "^4.1.0",
 		"keyv": "^4.5.0",
@@ -45,7 +46,6 @@
 		"@types/create-test-server": "^3.0.1",
 		"@types/delay": "^3.1.0",
 		"@types/get-stream": "^3.0.2",
-		"@types/http-cache-semantics": "^4.0.1",
 		"@types/jest": "^29.0.3",
 		"@types/node": "^18.7.18",
 		"@types/responselike": "^1.0.0",

